src/shared/hooks/useInterval.ts

```diff
@@ -0,0 +1,21 @@
+import React, { useState, useEffect, useRef } from "react";
+
+// creating the custom useInterval hook
+export function useInterval (callback: Function, delay: number): void {
+  // Creating a ref
+  const savedCallback = useRef<Function | null>(null);
```

Comment:
...Don't use `Function` as a type. The `Function` type accepts any function-like value...
If u see some errors for TS, try to fix them.
In this case we can create:

    type CallbackFunction = () => void;

And use CallbackFunction instead of Function.